curriculum/challenges/english/25-front-end-development/lab-factorial-calculator/66c07238b01053abaf812065.md

```diff
@@ -0,0 +1,48 @@
+---
+id: 66c07238b01053abaf812065
+title: Build a Factorial Calculator
+challengeType: 14
+dashedName: build-a-factorial-calculator
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should ask the user for a number using a prompt window and store the number in a variable called `inputNumber`.
+   
+2. Declare a variable `result` and assign it the value of `1`.
+   
+3. Using a for loop, you should loop through all numbers from `1` to `inputNumber` (inclusive) and for each number, multiply the `result` variable by the current number and assign the result to the `result` variable.
+   
+4. You should log the result of the factorial of `inputNumber` to the console. The output should be in the format `The factorial of [inputNumber] is [result]`.
+
+# --hints--
+
+Your `link` element should be within your `head` element.
+
+```js
+
+```
+
+# --seed--
+
+## --seed-contents--
+
+```js
+
+```
+
+# --solutions--
+
+```js
+let inputNumber = prompt("Enter a number: ");
+let result = 1;
+for (let i = 1; i <= inputNumber; i++) {
+  result *= i;
+}
+
+console.log(`Factorial of ${inputNumber} is: ${result}`); 
```

Comment:
I was playing around with this locally and I noticed a few things missing.

Technically the user can input anything

<img width="950" alt="Screenshot 2024-08-19 at 11 21 48 AM" src="https://github.com/user-attachments/assets/22532e62-7e71-401f-b43a-7ecd105ad6c5">

But also, prompt will return a string
<img width="952" alt="Screenshot 2024-08-19 at 11 22 44 AM" src="https://github.com/user-attachments/assets/010bf3c7-9af4-4d9f-85e5-4c4f7b22347e">

So this solution as well as the user stories will need to be updated to check for the following:

- we should have campers write an condition to check that the input is not an empty string
- then campers will need to convert the input to a number and check that it is not NaN

<img width="844" alt="Screenshot 2024-08-19 at 11 26 30 AM" src="https://github.com/user-attachments/assets/00dfb2d5-c22b-4e61-9653-8c44b71c30c9">

if the user does not provide a valid input for either of those conditions above, then they should be prompted again until they do provide a valid input.

Reply:
campers will also know about isNaN at that point if you wanted them to check using that